plugin/auth_ldap/src/connection.cc

```diff
@@ -5,20 +5,87 @@
 
 #include "plugin/auth_ldap/include/plugin_log.h"
 
+namespace {
+// example of this callback is in the OpenLDAP's
+// servers/slapd/back-meta/bind.c (meta_back_default_urllist)
+int cb_urllist_proc(LDAP *ld, LDAPURLDesc **urllist, LDAPURLDesc **url,
+                    void *params) {
+  if (urllist == url) return LDAP_SUCCESS;
+
+  LDAPURLDesc **urltail;
+  for (urltail = &(*url)->lud_next; *urltail; urltail = &(*urltail)->lud_next)
```

Comment:
here, in case of ldap_host been out, and come back, we will always perform fail-back which is not always desired, like when the old primary gets back and it's still not in sync. MySQL connecting to it might allow revoked user to connect and so on.
I would rather make this behavior configurable.

Reply:
Even if we make it configurable, we are also using a connection pool. We either implement a mechanism that forces all of them to reconnect when we detect that the primary is back, or they remain connected to the current server.

Also, for your suggestion, we have two options:
* either leave the URL-list as is, making all connection attempts slower (considerably so, if the ldap port doesn't report a connection refused immediately)
* or implement a custom logic that tries to detect if the primary comes back alive, and modify the urllist then

These are doable, but I'm not sure if it's worth it, all examples I could find in opensource project used this same callback.

Reply:
I left this as-is for now, mostly because of what I've written above.

Reply:
Ack, maybe we just need to document the behavior as this is normally something people want to know when we have fail over capabilities.